lib/src/main/java/com/auth0/jwt/creators/AccessJwtCreator.java

```
@@ -126,7 +114,13 @@ public AccessJwtCreator withExp(Date exp) {
      * @throws IllegalArgumentException if the name is null.
      */
     public AccessJwtCreator withNonStandardClaim(String name, String value) {
-        jwt.withNonStandardClaim(name, value);
+        if(name.equalsIgnoreCase("subject") || name.equalsIgnoreCase(Claims.SUBJECT)) {
```

Let's use only the constant defined in Claims. No need to compare against a different value if you stop using it internally. Also flip the condition to avoid NPE

---

but whats stopping someone from calling .withNonStandardClaim("subject", "subject")?  i can do that straight in the unit test.  you want to throw an exception?